src/celengine/refcounted.h

```diff
@@ -0,0 +1,26 @@
+
+#pragma once
+
+class RefCounted
+{
+ protected:
+    mutable int m_refCount { 0 };
+    virtual ~RefCounted() = 0;
+ public:
+    int addRef() const
```

Comment:
This method modifies internal state, why it's `const`?

Reply:
Again, I just followed original implementation :-)